dsgrid/api/app.py

```diff
@@ -63,7 +63,9 @@
     raise Exception("The environment variable DSGRID_API_SERVER_STORE_DIR must be set.")
 offline_mode = True
 no_prompts = True
-spark = init_spark("dsgrid_api")
+# There could be collisions on the only-allowed SparkSession between the main process and
+# subprocesses that run queries.
+spark = init_spark("dsgrid_api", check_env=False)
```

Comment:
The default Hive metastore uses Apache Derby which only allows one user. Also, our standalone cluster on Eagle currently only allows one user (that could be improved if we are more particular about allocating cores). So, if the main process on the API server creates a session, receives a request to run a query, starts a child process which will also try to create a session, it will be deadlocked.